sycl/test/regression/usm_free.cpp

```diff
@@ -0,0 +1,50 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+
+//==-------------- usm_free.cpp - SYCL USM free malloc_shared and free test -------------==//
+//
+// This test checks if users will successfully allocate 160, 0, and -16 bytes of shared
+// memory, and also test user can call free() without worrying about nullptr or invalid
+// memory descriptor returned from malloc.
+//==-------------------------------------------------------------------------------------==//
+
+#include <iostream>
+#include <CL/sycl.hpp>
+#include <stdlib.h>
+using namespace cl::sycl;
+
+int main(int argc, char * argv[]) {
+
+
+  int rows = 5, rank = 2;
+
+  auto exception_handler = [](cl::sycl::exception_list exceptions) {
+        for (std::exception_ptr const &e : exceptions) {
+            try {
+                std::rethrow_exception(e);
+            }
+            catch (cl::sycl::exception const &e) {
+                std::cout << "Caught asynchronous SYCL "
+                             "exception during sparse::trsv:\n"
+                          << e.what() << std::endl;
+            }
+        }
+    };
+
+  cl::sycl::device dev = cl::sycl::device(cl::sycl::gpu_selector());
```

Comment:
This should not use a device specific selector so it can be tested on multiple types of devices.